### ISSUE TITLE
-[__NSDictionaryM setObject:forKey:]: key cannot be nil

### ISSUE THREAD

**Issue by steelzeh** [iZettle/sdk-ios#296]
# Expected Behavior

Taking a payment while the SDK does not crash.

# Current Behavior

The SDK crashes.

# Steps to Reproduce

Try to make a payment through the iZettle SDK

## Context 

* Operating Version: iOS 13.5.1
* SDK version: 2.5.3

## Failure Logs

![Screenshot 2020-06-16 at 15 17 33](https://user-images.githubusercontent.com/7465835/84779665-f8374f00-afe4-11ea-9abc-38b5be0eb722.png)

```
Thread 0 [Crashed]:

0    CoreFoundation                           0x1a771b5f0     _exceptionPreprocess + 224
1    libobjc.A.dylib                          0x1a743dbcc     objc_exception_throw + 55
2    CoreFoundation                           0x1a77710cc     _CFThrowFormattedException + 111
3    CoreFoundation                           0x1a777a6e0     -[__NSDictionaryM setObject:forKey:].cold.1 + 47
4    CoreFoundation                           0x1a75f7f80     -[__NSDictionaryM setObject:forKey:] + 931
5    iZettlePayments                          0x103ce375c     _hidden#6115_ (__hidden#6132_:71)
6    iZettlePayments                          0x103ce3518     _hidden#6114_ (__hidden#6132_:46)
7    iZettlePayments                          0x103c88e50     _hidden#2681_ (__hidden#2760_:55)
8    iZettleShared                            0x103e15dfc     _hidden#4503_ (__hidden#4567_:320)
9    iZettleShared                            0x103e157f8     _hidden#4493_ (__hidden#4567_:261)
10   iZettleShared                            0x103e1572c     _hidden#4492_ (__hidden#4567_:297)
11   iZettleShared                            0x103e147b4     _hidden#4469_ (__hidden#4567_:122)
12   iZettleShared                            0x103e153dc     _hidden#4489_ (__hidden#4567_:259)
13   iZettleShared                            0x103e15d84     _hidden#4502_ (__hidden#4567_:316)
14   iZettlePayments                          0x103c88b48     _hidden#2679_ (__hidden#2760_:51)
15   iZettlePayments                          0x103ca44e0     _hidden#3693_ (__hidden#3717_:56)
16   iZettlePayments                          0x103cbc2ec     _hidden#4599_ (__hidden#4639_:45)
17   iZettleSDK                               0x103b2585c     _hidden#27_ (__hidden#114_:239)
18   iZettleSDK                               0x103b25290     _hidden#10_ (__hidden#114_:198)
19   iZettlePayments                          0x103cbd338     _hidden#4617_ (__hidden#4639_:134)
20   iZettleShared                            0x103e15dfc     _hidden#4503_ (__hidden#4567_:320)
21   iZettleShared                            0x103e157f8     _hidden#4493_ (__hidden#4567_:261)
22   iZettleShared                            0x103e1572c     _hidden#4492_ (__hidden#4567_:297)
23   iZettleShared                            0x103e147b4     _hidden#4469_ (__hidden#4567_:122)
24   iZettleShared                            0x103e153dc     _hidden#4489_ (__hidden#4567_:259)
25   iZettleShared                            0x103e15d84     _hidden#4502_ (__hidden#4567_:316)
26   iZettlePayments                          0x103cbcd4c     _hidden#4611_ (__hidden#4639_:113)
27   iZettleShared                            0x103dfe0fc     _hidden#3512_ (__hidden#3618_:171)
28   iZettleShared                            0x103dfe3f8     _hidden#3518_ (__hidden#3618_:188)
29   iZettlePayments                          0x103cbcab0     _hidden#4610_ (__hidden#4639_:99)
30   iZettleShared                            0x103e0f1d8     _hidden#4257_ (__hidden#4267_:114)
31   iZettlePayments                          0x103cbc8f4     _hidden#4609_ (__hidden#4639_:95)
32   iZettleSDK                               0x103b24f7c     _hidden#9_ (__hidden#114_:194)
33   iZettleSDK                               0x103b256b0     _hidden#26_ (__hidden#114_:231)
34   REDACTED                                 0x1025fc23c     closure #1 in PaymentViewController.chargeIzettle(payment:split:tips:) (PaymentViewController.swift:1748)
35   REDACTED                                 0x102724af0     thunk for @escaping @callee_guaranteed () -> () (<compiler-generated>:0)
36   libdispatch.dylib                        0x1a73e09a8     _dispatch_call_block_and_release + 23
37   libdispatch.dylib                        0x1a73e1524     _dispatch_client_callout + 15
38   libdispatch.dylib                        0x1a73935b4     _dispatch_main_queue_callback_4CF$VARIANT$mp + 903
39   CoreFoundation                           0x1a7699748     _CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 11
40   CoreFoundation                           0x1a769461c     _CFRunLoopRun + 1723
41   CoreFoundation                           0x1a7693c34     CFRunLoopRunSpecific + 423
42   GraphicsServices                         0x1b17dd38c     GSEventRunModal + 159
43   UIKitCore                                0x1ab7c622c     UIApplicationMain + 1931
44   libdyld.dylib                            0x1a751b800     $start + 3
```

**Comment by dmilovanovi** [comment]
Hi,

Thank you for reporting this issue.

Apologies for the late reply and for any inconvenience this may have caused.

We are currently investigating what is causing the SDK to crash when making payments and will get back to you once we have more information.

Best,
Dejan

**Comment by valkovsky** [comment]
Hi,

We identified the reason for this crash and prepared a fix for it. It will be a part of 3.0 release that we gonna release soon.
You need to update SDK version to 3.0, when it's released, in order to resolve this problem.

**Comment by dmilovanovi** [comment]
Hi @steelzeh ,

SDK 3.0 has been released and is now available at https://github.com/iZettle/sdk-ios/releases/tag/3.0.0.

Please have a look and let me know if you have any questions.

Regards,
Dejan

**Comment by sammcewan** [comment]
Our customers are getting heaps of these issues. It needs fixing in 2.0

As a warning to others, to try and resolve this we followed the above advice and put a ton of effort into upgrading to 3.0.0 and then found out there was a significant oversight in the auth process on iZettle's behalf that meant we then had to roll back our customers.

Can we get this fixed in the 2.0 SDK because there's no way we're going near 3.0 for the foreseeable future!

**Comment by dmilovanovi** [comment]
Hey Sam,

I've sent you an email the other day regarding v3.0.0 and staff + demo accounts not being able to log in.

Just wanted to let you know that the auth issue has been fixed and merchants should no longer have these issues.

We will also release v3.0.1 soon with more fixes and stable improvements.

Please accept my sincere apologies for the inconvenience and let me know if you have any questions.

I'll also make sure to prioritize your emails.

Regards,
Dejan
